pandas/core/indexes/base.py

```diff
@@ -3931,6 +3933,17 @@ def __getitem__(self, key):
         key = com.values_from_object(key)
         result = getitem(key)
         if not is_scalar(result):
+            if np.ndim(result) > 1:
+                # GH#27125 indexer like idx[:, None] expands dim, but we
+                #  cannot do that and keep an index, so return ndarray
+                # Deprecation GH#30588
+                warnings.warn(
+                    "Support for Index[:, None] is deprecated and will be "
+                    "removed in a future version.",
```

Comment:
Same comment here about `Index[:, None]` and recommended alternative as I posted above for the whatsnew